pkg/mimirtool/client/client.go

```diff
@@ -174,36 +173,34 @@ func checkResponse(r *http.Response) error {
 		return nil
 	}
 
-	var msg, errMsg string
-	scanner := bufio.NewScanner(io.LimitReader(r.Body, 512))
-	if scanner.Scan() {
-		msg = scanner.Text()
+	bodyHead, err := io.ReadAll(io.LimitReader(r.Body, 1024))
+	if err != nil {
+		return errors.Wrapf(err, "reading body")
 	}
 
+	msg := string(bodyHead)
+	var errMsg string
 	if msg == "" {
-		errMsg = fmt.Sprintf("server returned HTTP status %s", r.Status)
+		errMsg = fmt.Sprintf("server returned HTTP status: %s", r.Status)
 	} else {
-		errMsg = fmt.Sprintf("server returned HTTP status %s: %s", r.Status, msg)
+		errMsg = fmt.Sprintf("server returned HTTP status: %s, body: %s", r.Status, msg)
 	}
 
 	if r.StatusCode == http.StatusNotFound {
 		log.WithFields(log.Fields{
 			"status": r.Status,
-			"msg":    msg,
 		}).Debugln(errMsg)
 		return ErrResourceNotFound
 	}
 	if r.StatusCode == http.StatusConflict {
 		log.WithFields(log.Fields{
 			"status": r.Status,
-			"msg":    msg,
 		}).Debugln(errMsg)
 		return errConflict
 	}
 
 	log.WithFields(log.Fields{
 		"status": r.Status,
-		"msg":    msg,
```

Comment:
Why did this change?  I don't see it mentioned in the commit description.

Reply:
it doubles the error message. `errMsg` already contains `msg`

https://github.com/grafana/mimir/blob/465cbffd1a8b11841c003c86d03ac548e9cd847f/pkg/mimirtool/client/client.go#L180-L194

Reply:
I noticed the same as @dimitarvdimitrov is saying, figured that's why `msg` gets removed.

Reply:
actually I went ahead and did some more changes. even with the removed `msg` (and with `%q`), there was still duplication

```
time="2022-10-11T07:28:28Z" level=error msg="server returned HTTP status: 400 Bad Request, body: \"block upload is disabled\\n\"" status="400 Bad Request"
```

in da755e9c073951a7eef0a40f6b86e13c021d1046 I changed this to 

```
time="2022-10-11T07:45:02Z" level=error msg=response body="block upload is disabled\n" status="400 Bad Request"
```

and an example flow is for a failed blocks upload

```
time="2022-10-11T07:45:02Z" level=info msg=Backfilling blocks=/shared/01GF301Z9T3CX0YK0VZ5B18MWJ user=anonymous
time="2022-10-11T07:45:02Z" level=info msg="making request to start block upload" block=01GF301Z9T3CX0YK0VZ5B18MWJ file=meta.json path=/shared/01GF301Z9T3CX0YK0VZ5B18MWJ
time="2022-10-11T07:45:02Z" level=error msg=response body="block upload is disabled\n" status="400 Bad Request"
time="2022-10-11T07:45:02Z" level=error msg="failed uploading block" error="request to start block upload failed: POST request to http://compactor:8080/api/v1/upload/block/01GF301Z9T3CX0YK0VZ5B18MWJ/start failed: server returned HTTP status: 400 Bad Request, body: \"block upload is disabled\\n\"" path=/shared/01GF301Z9T3CX0YK0VZ5B18MWJ
time="2022-10-11T07:45:02Z" level=info msg="finished uploading blocks" already_exists=0 failed=1 succeeded=0
mimirtool: error: blocks failed to upload 1 block(s), try --help
```